medmodels/medrecord/medrecord.py

```diff
@@ -591,34 +591,57 @@ def add_nodes(
             PolarsNodeDataFrameInput,
             List[PolarsNodeDataFrameInput],
         ],
+        group: Optional[Group] = None,
     ) -> None:
-        """Adds multiple nodes to the MedRecord from different data formats.
+        """Adds multiple nodes to the MedRecord from different data formats and optionally assigns them to a group.
 
         Accepts a list of tuples, DataFrame(s), or PolarsNodeDataFrameInput(s) to add
         nodes. If a DataFrame or list of DataFrames is used, the add_nodes_pandas method
         is called. If PolarsNodeDataFrameInput(s) are provided, each tuple must include
-        a DataFrame and the index column.
+        a DataFrame and the index column. If a group is specified, the nodes are added
+        to the group.
 
         Args:
             nodes (Union[Sequence[NodeTuple], PandasNodeDataFrameInput, List[PandasNodeDataFrameInput], PolarsNodeDataFrameInput, List[PolarsNodeDataFrameInput]]):
                 Data representing nodes in various formats.
+            group (Optional[Group]): The name of the group to add the nodes to. If not
+                specified, the nodes are added to the MedRecord without a group.
 
         Returns:
             None
         """
-        if is_pandas_node_dataframe_input(nodes) or is_pandas_node_dataframe_input_list(
-            nodes
-        ):
-            return self.add_nodes_pandas(nodes)
-        elif is_polars_node_dataframe_input(
-            nodes
-        ) or is_polars_node_dataframe_input_list(nodes):
-            return self.add_nodes_polars(nodes)
+        nodes_indices = []
+
+        if is_pandas_node_dataframe_input(nodes):
+            nodes_indices = nodes[0][nodes[1]].to_list()
+            self.add_nodes_pandas(nodes)
+        elif is_pandas_node_dataframe_input_list(nodes):
+            nodes_indices = [
+                nodes for node in nodes for nodes in node[0][node[1]].tolist()
+            ]
+            self.add_nodes_pandas(nodes)
+
+        elif is_polars_node_dataframe_input(nodes):
+            nodes_indices = nodes[0][nodes[1]].to_list()
+            self.add_nodes_polars(nodes)
+        elif is_polars_node_dataframe_input_list(nodes):
+            nodes_indices = [
+                nodes for node in nodes for nodes in node[0][node[1]].to_list()
+            ]
+            self.add_nodes_polars(nodes)
+
         else:
-            return self._medrecord.add_nodes(nodes)
+            nodes_indices = [node[0] for node in nodes]
+            self._medrecord.add_nodes(nodes)
+
+        if group is not None:
+            if not self.contains_group(group):
+                self.add_group(group)
+            self._medrecord.add_node_to_group(group, nodes_indices)
```

Comment:
this calculates the node indices even when a group is not set which might be an expensive operation